batch/batch/front_end/front_end.py

```diff
@@ -352,13 +398,13 @@ async def _get_job_log_from_record(app, batch_id, job_id, record):
 
     if state == 'Running':
         try:
-            resp = await request_retry_transient_errors(
-                client_session, 'GET', f'http://{ip_address}:5000/api/v1alpha/batches/{batch_id}/jobs/{job_id}/log'
-            )
-            return await resp.json()
+            resp = await request_retry_transient_errors(client_session, 'GET', f'http://{ip_address}:5000{endpoint}')
+            return await handle_running_response(resp)
         except aiohttp.ClientResponseError:
-            log.exception(f'while getting log for {(batch_id, job_id)}')
-            return {task: 'ERROR: encountered a problem while fetching the log' for task in tasks}
+            log.exception(f'while getting resource for {(batch_id, job_id)}')
+            return {
+                task: GetResourceError('ERROR: encountered a problem while fetching the resource') for task in tasks
+            }
```

Comment:
This abstraction feels a little clunky and like it's not strictly beneficial. I think this is a situation where saving lines of code came at the sacrifice of local interpretability. As an example, let's say I see this error in the logs (which now just say "getting resource" instead of "getting log", which is less informative). If I want to understand what happened such that line number 404 triggered, I have to look at the invocation of this function to see which `endpoint` was used, then jump back here to see how it was used, then immediately jump back to the invocation to see what `handle_running_response` is. I think that this abstraction was too surgical instead of a natural way to break up responsibilities.

Reply:
I also think we should get rid of `GetResourceError`.

```
'ERROR: encountered a problem while fetching the resource'
```

This is what the user will see if we can't get the job log, which is going to be less intuitive than a message saying we couldn't get the log. I feel like this method should return `None` for any task that it can't find, and the client of this method should figure out how to interpret missing items.

Reply:
I feel like this should just be broken up into separate functions: one that communicates with the worker and returns the `ClientResponse` that the callee can then process however it wants, and another which reads from cloud storage. The two callees (getting log file and getting resource file) will have to duplicate the `if state == 'running'`, but that seems a small duplication and I think will significantly help clarity as then we won't need this `handle_running_response` input.